libr/bin/format/mach0/mach0.c

```diff
@@ -1679,6 +1679,21 @@ int MACH0_(get_bits)(struct MACH0_(obj_t)* bin) {
 #endif
 }
 
+int MACH0_(get_bits_from_hdr)(struct MACH0_(mach_header)* hdr) {
+	#if R_BIN_MACH064
+		return 64;
+	#else
+		//this hack only applies with ARM cpu 
+		if (hdr->cputype == CPU_TYPE_ARM) {
```

Comment:
this is wrong. I know you I know you copied the code from mach0.c but there, there is `bin->entry & 1` to see if the code is thumb or arm, but in this case you are missing that check so everything will be recognized as thumb with CPU_TYPE_ARM. The entrypoint in mach doesn't reside in the header instead in a command of type LC_MAIN that to parse for 999 files seems to me overkill. I need to think how to cope with this. Any idea?